File: ai/mill-ai-core/src/main/java/io/qpointz/mill/ai/capabilities/Capability.java
package io.qpointz.mill.ai.capabilities;

import org.springframework.ai.chat.client.advisor.api.Advisor;
import org.springframework.ai.tool.ToolCallback;

import java.util.List;

public interface Capability {

    List<Advisor> getAdvisors();

    List<Object> getTools();

    List<ToolCallback> getToolCalbacks();


}

File: ai/mill-ai-v2/src/main/kotlin/io/qpointz/mill/ai/Capability.kt
package io.qpointz.mill.ai

import com.fasterxml.jackson.databind.JsonNode
import org.springframework.ai.chat.client.advisor.api.Advisor
import org.springframework.ai.tool.ToolCallback


interface Capability {
    val name: String
    val description: String
    val system: String?
    val tools: List<ToolCallback>
    val advisors: List<Advisor>
    val protocol: JsonNode?
}
